Inventory.cs

```diff
@@ -84,9 +86,9 @@ public void AddProduct()
         {
             try
             {
-                string name = GetValidatedInput("Enter product name: ", "^[a-zA-Z0-9 ]+$", true, 50);
-                decimal price = GetValidatedDecimal("Enter product price: ");
-                int quantity = GetValidatedInt("Enter product quantity: ");
+                string name = GetValidatedInput<string>("Enter product name: ", "^[a-zA-Z0-9 ]+$", true, 50);
```

Comment:
you should place the regex patterns in a static class called RegexHelper for example